conmon/conmon.c

```diff
@@ -429,26 +496,8 @@ int main(int argc, char *argv[])
 
 					ninfo("read a chunk: (fd=%d) '%s'", mfd, buf);
 
-
-					/* Prepend the CRI-mandated timestamp and other metadata. */
-					/*
-					 * FIXME: Currently this code makes the assumption that
-					 *        @buf doesn't contain any newlines (since the CRI
-					 *        requires each line to contain a timestamp). This
-					 *        is an /okay/ assumption in most cases because
-					 *        ptys generally have newline-buffered output.
-					 */
-					int rc = set_k8s_timestamp(tsbuf, TSBUFLEN, "stdout");
-					if (rc < 0) {
-						nwarn("failed to set timestamp");
-					} else {
-						if (write(logfd, tsbuf, TSBUFLEN) != TSBUFLEN) {
-							nwarn("partial/failed write ts (logFd)");
-						}
-					}
-					/* Log all output to logfd. */
-					if (write(logfd, buf, num_read) != num_read) {
-						nwarn("partial/failed write (logFd)");
+					if (write_k8s_log(logfd, "stdout", buf, num_read) < 0) {
```

Comment:
Can you verify whether the actual log file has incorrect data in it, or just the `stderr` log from `conmon`? Because to be honest at the moment `ninfo` is currently reading out-of-bounds and we need to stop doing that, so I hope that's the only issue here. :wink: